### PR TITLE
Introduce Vault component protected with Intel Secure Guard Extensions (SGX)

### PR DESCRIPTION
**Pull request by tsadowsk**

### CODE REVIEW

File: content/patterns/multicloud-gitops-sgx/_index.adoc [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+==> Vault server started! Log data will stream in below:
+
+...
+2024-02-26T15:59:57.791Z [INFO]  events: Starting event system
+```
+
+[NOTE]
+====
+This is a sample usage of the SGX application, as it is, and cannot be used in production; unless the above warnings related to `sgx.allowed_files` related to Gramine are resolved.
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx/_index.adoc]: please change `cannot` to `should not` ;)

File: content/patterns/multicloud-gitops-sgx/mcg-sgx-demo-script.adoc [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+. Log into the Vault using root token. Root token can be found by executing command:
++
+[source,terminal]
+----
+oc -n imperative get secrets vaultkeys -ojsonpath='{.data.vault_data_json}' | base64 -d
+----
++
+
+. After login go to `secret` catalog and clik *Create secret* and fill all the fields manually (Figure 2):
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx/mcg-sgx-demo-script.adoc]: When logged in, go to the `secret` catalog and clik *Create secret*. Fill all the fields manually (Figure 2):

File: content/patterns/multicloud-gitops-sgx/mcg-sgx-demo-script.adoc [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+.. *Path for this secret* is `global/config-demo` (from _values.yaml_ file for `config-demo` charts)
+
+.. Under *Secret data* key is `secret` (from _values-secret-multicloud-gitops.yaml_ file) and in next field put its value.
+
+.. Click *Add* and then *Save*.
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx/mcg-sgx-demo-script.adoc]: TBH I don't get it... Let me try to rephrase it, and decide if this is clear:

a) Put `global/config-demo` value in the _Path for this secret_ field (the value cames from _values-secret-multicloud-gitops.yaml_ file).
b) Add one _Secret data_ key-value pair. Put `secret` as a key (left field). The value must acquired from _values-secret-multicloud-gitops.yaml_ file. Click _Add_ button to confirm.
c) Click _Save_ to save changes.

THIS IS HOW I UNDERSTAND THIS PART - PLEASE VALIDATE A LOGIC BEFORE REPLACING YOUR PART ;)

**BTW is this section twice? In the "Getting started" page, and in the "Demo Script"?**

**Review by pgrabusz — CHANGES_REQUESTED** [review]
Good job Tomasz! :)
Small comments here and there, + those:
1) The image does not load 
image
2) I couldn't find it in the files, but in the `Getting started` section, there's a sentence `This config file can be adjusted as needed` with no dot at the end ;)
3) In the `Cluster sizing` tab, please change recommended cluster size:
  * Control Planes CPU: **2x** 5th Gen Intel® Xeon® Gold 6526Y (16 cores at 2.8 GHz base, SGX 512 GB) or better
  * Workers CPU: **2x** 5th Gen Intel® Xeon® Gold 6538Y+(32 cores at 2.2 GHz base, SGX 512 GB) or better
4) In the `Cluster sizing` tab, please change the "tested in the on-premises environment with following HW" table to:
  * CP + W CPU: **2x** 5th Gen Intel® Xeon® Platinum 8568Y+ (48 cores at 2.3 GHz base, SGX 512 GB)
  * CP + W Memory: 512 GB (16x32GB DDR5 5600)
 5) I saw that the section describing adding secrets to Vault is included twice - in the _demo script_ page, and in the _getting started_ page. Is it on purpose?

**Comment by pgrabusz** [comment]
One more thing...
there's a section describing a cluster deployment in the `Getting started` section.
It starts with `After succeeded installation open Validated Patterns Operator`...
I think we should rephrase it too. Let's try sth like:
```
After a successful installation, open the Validated Patterns Operator page. Next, go to the Pattern tab and click Create Pattern.
1. Set the _Name_ field to `multicloud-gitops-amx-rhoai`, and the _Cluster Group Name_ to `hub`. Values must be the same as in the _values-global.yaml_ file (Figure 3).
2. As a *Git Config > Target Repo* value, paste the link to your fork. Under *Git Config > Target Revision* write the name of your branch (Figure 3).
3. Click the _Create_ button to create the pattern.
```

File: content/patterns/multicloud-gitops-sgx/_index.adoc (hunk 1 above)
Comment (tsadowsk) [reply content/patterns/multicloud-gitops-sgx/_index.adoc]: changed

File: content/patterns/multicloud-gitops-sgx/mcg-sgx-demo-script.adoc (hunk 2 above)
Comment (tsadowsk) [reply content/patterns/multicloud-gitops-sgx/mcg-sgx-demo-script.adoc]: This section was removed from `Demo Script` page.

File: content/patterns/multicloud-gitops-sgx/mcg-sgx-demo-script.adoc (hunk 3 above)
Comment (tsadowsk) [reply content/patterns/multicloud-gitops-sgx/mcg-sgx-demo-script.adoc]: This section was removed from Demo Script page.

**Comment by tsadowsk** [comment]
I added rest of the mentioned changes.
